modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -438,7 +438,7 @@ private void exec() {
                     if (entity.getType().getTag() == TypeTags.XML_TAG) {
                         sf.longRegs[j] = ((BXML) entity).length();
                         break;
-                    } else if (entity.getType().getTag() == TypeTags.JSON_TAG) {
+                    } else if (entity instanceof BJSON) {
```

Comment:
Can you please explain the need to use the "instanceof" operator here?

Reply:
With the new change (to fix #4142), a type of a json (BJSON) can be one of: int,string,float,boolean, json[] or json. So just by looking at the type, we can't say whether its a json (BJSON). Hence the instanceof check.